creator-node/src/diskManager.ts

```diff
@@ -561,3 +565,160 @@ export async function sweepSubdirectoriesInFiles(
   // keep calling this function recursively without an await so the original function scope can close
   if (redoJob) return sweepSubdirectoriesInFiles()
 }
+
+async function _copyLegacyFiles(
+  legacyPathsAndCids: { storagePath: string; cid: string }[],
+  deleteLegacyFiles: boolean,
+  logger: Logger
+): Promise<
+  {
+    legacyPath: string
+    nonLegacyPath: string
+  }[]
+> {
+  const erroredPaths: { [path: string]: [error: string] } = {}
+  const copiedPaths: {
+    legacyPath: string
+    nonLegacyPath: string
+  }[] = []
+  for (const { storagePath: legacyPath, cid } of legacyPathsAndCids) {
+    try {
+      // Compute new path
+      const nonLegacyPathInfo = extractCIDsFromFSPath(legacyPath)
+      if (!nonLegacyPathInfo) throw new Error('Unable to extract CID from path')
+      if (!nonLegacyPathInfo.inner && !nonLegacyPathInfo.outer) {
+        throw new Error('Extracted empty CID from path')
+      }
+
+      // Ensure new path's parent exists
+      const nonLegacyPath = await (nonLegacyPathInfo.isDir
+        ? computeFilePathInDirAndEnsureItExists(
+            nonLegacyPathInfo.outer,
+            nonLegacyPathInfo.inner!
+          )
+        : computeFilePathAndEnsureItExists(nonLegacyPathInfo.outer))
+
+      // Copy to new path if it doesn't already exist
+      if (await fs.pathExists(nonLegacyPath)) {
+        copiedPaths.push({ legacyPath, nonLegacyPath })
+      } else {
+        try {
+          await fs.copyFile(legacyPath, nonLegacyPath)
+        } catch (e: any) {
+          // If we see a ENOSPC error, log out the disk space and inode details from the system
+          if (e.message.includes('ENOSPC')) {
+            await Promise.all([
+              runShellCommand(`df`, ['-h'], logger),
+              runShellCommand(`df`, ['-ih'], logger)
+            ])
+          }
+          throw e
+        }
+      }
+
+      // Verify that the correct contents were copied
+      const cidMatchesExpected = await verifyCIDMatchesExpected({
```

Comment:
awesome - idk if this is better protection than just comparing md5 of the files at old and new paths, bc i guess the old file's cid could be wrong?

Reply:
fair